packages/eslint/src/generators/utils/eslint-file.ts

```diff
@@ -493,7 +497,7 @@ export function addExtendsToLintConfig(
       return addDependenciesToPackageJson(
         tree,
         {},
-        { '@eslint/compat': eslintCompat },
+        { '@eslint/compat': eslintCompat, '@eslint/eslintrc': eslintrcVersion },
```

Comment:
The `@eslint/eslintrc` package needs to be installed regardless of `shouldImportEslintCompat`. With the current change, it's only going to be installed if `shouldImportEslintCompat === true`.